src/HtsSystemContract.sol

```diff
@@ -160,33 +163,69 @@ contract HtsSystemContract is IHederaTokenService, IERC20Events {
      */
     function __redirectForToken() internal virtual returns (bytes memory) {
         bytes4 selector = bytes4(msg.data[24:28]);
+        bytes memory result = abi.encode(false);
+        _initTokenData();
+
+        // Redirect to the appropriate ERC20 method if the token type is fungible.
+        if (keccak256(bytes(tokenType)) == keccak256(bytes("FUNGIBLE_COMMON"))) {
+            result = __redirectForERC20(selector);
+            if (keccak256(result) != keccak256(abi.encode("undefined"))) {
```

Comment:
In the unlikely case say for example `name` or `symbol` is `"undefined"`, will this work as expected?

Reply:
That's right - it wouldn't, perhaps a hash of "undefined" would work better since it's less likely to appear in the actual field

Reply:
I will try to emove the "undefined" results and use a modifier for __redirectForHRC719